src/app.rs

```diff
@@ -198,9 +200,14 @@ fn draw_fan_speed<'d>(gpu: &GpuInfo<'d>) -> Gauge<'d> {
         .map(|u| u as f64)
         .sum::<f64>()
         / temps as f64;
-
-    let percentage = (avg / 100.).clamp(0., 1.0);
-    let label = format!("{:.1}%", avg);
+    let percentage = match avg.is_nan() {
```

Comment:
There is defnitely no issue with the code  -- I'm thinking that maybe we should just not run the call to `draw_fan-speed` at all if we're on a fan-less system, because we're checking this every time the loop runs. 

Wanna hoist something out side the loop to where we can check if we get values then just never run this code if we don't have to?

Reply:
i will convert it to something like Result<Gauge, Error>, if error is no fans, it will get there is no fans. and, you can implement other errors in future.